pallets/connectors/src/lib.rs

```diff
@@ -0,0 +1,522 @@
+
+#[frame_support::pallet]
+pub mod pallet {
+	use cfg_primitives::Moment;
+	use cfg_traits::{Permissions, PoolInspect};
+	use cfg_types::{PermissionScope, PoolRole, Role};
+	use frame_support::{error::BadOrigin, pallet_prelude::*, traits::UnixTime};
+	use frame_system::pallet_prelude::*;
+	use pallet_xcm_transactor::{Currency, CurrencyPayment, TransactWeights};
+	use sp_runtime::traits::{AccountIdConversion, Zero};
+	use xcm::latest::OriginKind;
+
+	use super::*;
+	use crate::weights::WeightInfo;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub (super) trait Store)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + pallet_xcm_transactor::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		type WeightInfo: WeightInfo;
+
+		type Balance: Parameter
+			+ Member
+			+ AtLeast32BitUnsigned
+			+ Default
+			+ Copy
+			+ MaybeSerializeDeserialize
+			+ MaxEncodedLen;
+
+		type Rate: Parameter + Member + MaybeSerializeDeserialize + FixedPointNumber + TypeInfo;
+
+		type CurrencyId: Parameter
+			+ Copy
+			// TODO(nuno): remove Default is not needed after MVP tests
+			+ Default
+			+ IsType<<Self as pallet_xcm_transactor::Config>::CurrencyId>;
+
+		type AdminOrigin: EnsureOrigin<Self::Origin>;
+
+		type PoolInspect: PoolInspect<
+			Self::AccountId,
+			<Self as pallet::Config>::CurrencyId,
+			Rate = Self::Rate,
+		>;
+
+		type Permission: Permissions<
+			Self::AccountId,
+			Scope = PermissionScope<PoolIdOf<Self>, <Self as pallet::Config>::CurrencyId>,
+			Role = Role<TrancheIdOf<Self>, Moment>,
+			Error = DispatchError,
+		>;
+
+		type Time: UnixTime;
+
+		type Tokens: Mutate<Self::AccountId>
+			+ Inspect<
+				Self::AccountId,
+				AssetId = <Self as pallet::Config>::CurrencyId,
+				Balance = <Self as pallet::Config>::Balance,
+			> + Transfer<Self::AccountId>;
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub (super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// A message was sent to a domain
+		MessageSent {
+			domain: Domain,
+			message: MessageOf<T>,
+		},
+
+		/// The Router for a given Domain was set
+		SetDomainRouter {
+			domain: Domain,
+			router: Router<CurrencyIdOf<T>>,
+		},
+	}
+
+	#[pallet::storage]
+	pub(crate) type DomainRouter<T: Config> =
+		StorageMap<_, Blake2_128Concat, Domain, Router<CurrencyIdOf<T>>>;
+
+	/// TMP
+	#[pallet::storage]
+	pub(crate) type FeeCurrency<T: Config> =
+		StorageMap<_, Blake2_128Concat, Domain, CurrencyIdOf<T>>;
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// A pool could not be found
+		PoolNotFound,
+		/// A tranche could not be found
+		TrancheNotFound,
+		/// The specified domain has no associated router
+		InvalidDomain,
+		/// Failed to send a message
+		SendFailure,
+		/// Token price is not set
+		MissingPrice,
+		/// The router does not exist
+		MissingRouter,
+		/// The selected domain is not yet supported
+		UnsupportedDomain,
```

Comment:
An edge case: in theory, at some point we could disable a router to some domain again. Then the Domain would still be included in the enum (since the indices need to remain consistent, similar to the CurrencyId enum), but it would throw this error.

Reply:
Ok cool, that's fair 👍

Reply:
I am adding the Domain variants back but keep this error variant deleted since we don't need to support this now.